packages/cli-repl/src/cli-repl.ts

```diff
@@ -605,6 +626,19 @@ export class CliRepl implements MongoshIOProvider {
     );
   }
 
+  async setLoggingEnabled(enabled: boolean): Promise<void> {
+    if (enabled) {
+      if (!this.logWriter) {
+        await this.startLogging();
+      }
+      // Do nothing if there is already an active log writer.
+    } else {
+      this.loggingAndTelemetry?.detachLogger();
+      this.logWriter?.destroy();
+      this.logWriter = undefined;
```

Comment:
So we create a new logger (and new logId) each time logging gets reenabled, which also means if someone disable and then enables logging, a new file is created.
I think that is reasonable unless we'd rather make sure it writes to the same file again (thinking of it more of a "pause")